### ISSUE TITLE
WebOS ext for VS Code incorrect between open from Ubuntu desktop icon and terminal

### ISSUE THREAD

**Issue by IRTonyCHAN** [webos-tools/webos-studio#5]
When I open VS Code from terminal "code", the WebOS ext everything is fine.
![image](https://github.com/user-attachments/assets/e3ea92ed-e83f-4a8d-84c3-59dc4967e6fd)

While I open through desktop icon, it keep loading and notice not installed dependencies
![image](https://github.com/user-attachments/assets/a41ad5ce-93d7-4d7e-9c11-594d39b6c658)

checked ~/.bashrc added the following line
export WEBOS_CLI_TV="/home/tony/.nvm/versions/node/v20.13.1/bin"
export WEBOS_CLI_TV="/usr/bin"

How to make the desktop icon working? Any step I missed?

/usr/share/applications/code.desktop
[Desktop Entry]
Name=Visual Studio Code
Comment=Code Editing. Redefined.
GenericName=Text Editor
Exec=/usr/share/code/code %F
Icon=vscode
Type=Application
StartupNotify=false
StartupWMClass=Code
Categories=TextEditor;Development;IDE;
MimeType=application/x-code-workspace;
Actions=new-empty-window;
Keywords=vscode;

[Desktop Action new-empty-window]
Name=New Empty Window
Name[de]=Neues leeres Fenster
Name[es]=Nueva ventana vacía
Name[fr]=Nouvelle fenêtre vide
Name[it]=Nuova finestra vuota
Name[ja]=新しい空のウィンドウ
Name[ko]=새 빈 창
Name[ru]=Новое пустое окно
Name[zh_CN]=新建空窗口
Name[zh_TW]=開新空視窗
Exec=/usr/share/code/code --new-window %F
Icon=vscode

**Comment by YoungeunKim** [comment]
Hi :) 
I searched for the issue and found a related page. Please refer to : 
https://code.visualstudio.com/docs/supporting/faq#_resolving-shell-environment-fails

I will check the issue further and let you know.

**Comment by IRTonyCHAN** [comment]
> Hi :) I searched for the issue and found a related page. Please refer to : https://code.visualstudio.com/docs/supporting/faq#_resolving-shell-environment-fails
> 
> I will check the issue further and let you know.

Thanks for reply, checked should be not related since never see that dialog showing.
![image](https://github.com/user-attachments/assets/1cf312b1-eb15-493c-b8e5-aa8d7fc578df)

When I try to open the application from the Ubuntu dock repeatedly, the ~/.bashrc file will append many lines.
export WEBOS_CLI_TV="/home/tony/.nvm/versions/node/v20.13.1/bin"
export WEBOS_CLI_TV="/usr/bin"
export WEBOS_CLI_TV="/home/tony/.nvm/versions/node/v20.13.1/bin"
export WEBOS_CLI_TV="/home/tony/.nvm/versions/node/v20.13.1/bin"
export WEBOS_CLI_TV="/home/tony/.nvm/versions/node/v20.13.1/bin"
export WEBOS_CLI_TV="/usr/bin"

I'm not sure if the issue is with the extension or with VS Code itself.
I will use the terminal to continue development at this moment.

**Comment by YoungeunKim** [comment]
Hi, 
I reproduced the issue on my ubuntu PC. 
The reason is that when VS Code was launched by shortcut, the terminal runs as a "non-login shell". So, the path settings in ~/.bashrc is not applied to the VS Code's terminal.


**Issue case**
When I launched VS Code by desktop icon, the installing global package pop-up was displayed.
The reason is that webOS Studio cannot find npm modules such as ares-cli : 
 (You can check `ares -V` command and` echo $PATH`)


![image](https://github.com/user-attachments/assets/56bc671a-3008-4783-bb56-d46426118ae1)

: ares-cli is not in path on the VS code's terminal. 

**how to fix**
Please refer to  https://github.com/Microsoft/vscode/issues/7263#issuecomment-224140256


I added the below lines to ~/.bash_profile and save the file.


```
# .bash_profile
if [ -r ~/.bashrc ]; then
  . ~/.bashrc
fi
```




Launch VS Code by shortcut. The installing pop-up is not showing. 
Check ares-cli version and path in the VS code's terminal :
![image](https://github.com/user-attachments/assets/bde496bc-6434-4472-a278-41f652ac2aa1)

Please check the ares-cli path with "which ares" and it should be in the PATH.(or globally installed npm directory path)

```
ye0607@ye0607Linux:~$ which ares
/home/ye0607/.npm-global/bin/ares
```

The npm path and ares-cli installation path depend on your environment settings. 
I hope this helps you solve the problem.

**Comment by IRTonyCHAN** [comment]
> Hi, I reproduced the issue on my ubuntu PC. The reason is that when VS Code was launched by shortcut, the terminal runs as a "non-login shell". So, the path settings in ~/.bashrc is not applied to the VS Code's terminal.
> 
> **Issue case** When I launched VS Code by desktop icon, the installing global package pop-up was displayed. The reason is that webOS Studio cannot find npm modules such as ares-cli : (You can check `ares -V` command and` echo $PATH`)
> 
> ![image](https://private-user-images.githubusercontent.com/5811141/356935754-56bc671a-3008-4783-bb56-d46426118ae1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM0Mzk5OTIsIm5iZiI6MTcyMzQzOTY5MiwicGF0aCI6Ii81ODExMTQxLzM1NjkzNTc1NC01NmJjNjcxYS0zMDA4LTQ3ODMtYmI1Ni1kNDY0MjYxMThhZTEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgxMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MTJUMDUxNDUyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YTJlMmE3YmNhOWNiNmYxYzk2Yzg2MjZjYWE1NzBlZmY5NTU2Yzc5ZGMxZjRjOGZlYTExZWY4NmU3YWRkODgyMCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.JL1FymMeuI0-K21fkeUoaBbqenw5673SsI92iYOmo1s)
> 
> : ares-cli is not in path on the VS code's terminal.
> 
> **how to fix** Please refer to [microsoft/vscode#7263 (comment)](https://github.com/microsoft/vscode/issues/7263#issuecomment-224140256)
> 
> I added the below lines to ~/.bash_profile and save the file.
> 
> ```
> # .bash_profile
> if [ -r ~/.bashrc ]; then
>   . ~/.bashrc
> fi
> ```
> 
> Launch VS Code by shortcut. The installing pop-up is not showing. Check ares-cli version and path in the VS code's terminal : ![image](https://private-user-images.githubusercontent.com/5811141/356934768-bde496bc-6434-4472-a278-41f652ac2aa1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM0Mzk5OTIsIm5iZiI6MTcyMzQzOTY5MiwicGF0aCI6Ii81ODExMTQxLzM1NjkzNDc2OC1iZGU0OTZiYy02NDM0LTQ0NzItYTI3OC00MWY2NTJhYzJhYTEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgxMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MTJUMDUxNDUyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDcyMTBjNWEyMjEzNzU3NmI3OWFjMjI2ZmU0MjI5NGM4YmM2NjQ1MmY5M2Y0YjlkMWQ1OTY3MDVhNjI3ZjBhMCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ._nyyjQbQ7LonzyRAnH41AR2Cq-H3VkpEbDizn6ocmBU)
> 
> Please check the ares-cli path with "which ares" and it should be in the PATH.(or globally installed npm directory path)
> 
> ```
> ye0607@ye0607Linux:~$ which ares
> /home/ye0607/.npm-global/bin/ares
> ```
> 
> The npm path and ares-cli installation path depend on your environment settings. I hope this helps you solve the problem.

The solution solved my issue. Thank you for your help.

**Comment by YoungeunKim** [comment]
good news :) 

I close this issue.
